src/Build/Instance/ProjectItemInstance.cs

```diff
@@ -1602,6 +1639,11 @@ public bool Equals(TaskItem other)
             /// </remarks>
             public bool HasMetadata(string name)
             {
+                if (string.IsNullOrEmpty(name))
+                {
+                    ErrorUtilities.VerifyThrowArgumentLength(name, nameof(name));
+                }
```

Comment:
Yes, I think it's necessary.  HasMetadata parameter and GetMetadata parameter should have the same limitation.

Reply:
That doesn't seem to be true. `_directMetadata?.Contains(name)` does not throw on null.

If there is a path that doesn't handle null, I think it would be reasonable to add a proper argument check here. I would be opposed to checking for an empty string, though, unless there is a good reason. The breaking potential is non-trivial.

Also, changes in the behavior of the public API should always come with the corresponding test coverage.

Reply:
> That doesn't seem to be true. `_directMetadata?.Contains(name)` does not throw on null.

If there is no null check. this will throw the exception as bellow.  I think this exception message is not  easier to undertand than ErrorUtilities.VerifyThrowArgumentLength(name, nameof(name));   And for empty string check, I think it's same with GetMetadata. but in order to avoid new  potential issues, I will only check null currently.

![image](https://user-images.githubusercontent.com/26814373/218954819-158db5a9-1ec1-422a-9777-9458d31cec32.png)

Reply:
Have added the tests for the new public API

Reply:
> The new `GetMetadataValueEscaped` still doesn't take empty string now. Do you think it would make sense to unify the behavior with `HasMetadata` and validate the argument so it throw only on null?

@Forgind  what’s your idea？